src/lib.rs

```diff
@@ -226,10 +226,15 @@ impl Rut {
     ///
     /// The Input must be a valid RUT format.
     pub fn from(input: &str) -> Result<Rut, Error> {
-        match Rut::extract_from(input) {
-            Ok(unverified_rut) => Rut::check_dv(unverified_rut),
-            Err(error) => Err(error),
+        if input.starts_with('0') {
+            let rut = Rut::extract_from(input.replace("0", "").as_str())?;
+
+            return Rut::check_dv(rut);
         }
+
+        let rut = Rut::extract_from(input)?;
+
+        Rut::check_dv(rut)
```

Comment:
To avoid extra allocations we are taking a different path here with +1 allocation when calling `input.replace(...`